src/data/set/function.lean

```diff
@@ -114,6 +114,14 @@ subtype.map f h
 @[simp] lemma maps_to.coe_restrict_apply (h : maps_to f s t) (x : s) :
   (h.restrict f s t x : β) = f x := rfl
 
+/-- Given a map `f` sending `univ : set α` into `t : set β`, restrict the codomain to `t`.
+Same as `subtype.coind`.-/
+def maps_to.restrict_codom {f : α → β} {t : set β} (h : maps_to f univ t) :
+  α → t := subtype.coind f (λ a, h (mem_univ a))
```

Comment:
```suggestion
def maps_to.cod_restrict {f : α → β} {t : set β} (h : maps_to f univ t) :
  α → t := t.cod_restrict f $ (t.maps_univ_to f).mp h
```
Since we already have `set.cod_restrict`

Reply:
Wait but then this is a duplicate... I did not know about `cod_restrict`...

Reply:
Ok it is exactly the same. This PR should be closed. Do you agree?

Reply:
There's still maybe an argument for having this `def` because it enables dot notation on `maps_to`.

Reply:
But then `cod_restrict` be refactored to this one (also `cod_restrict` has a lot of unnecessary explicit parameters that should be refactored). I actually believe we should give another name to `maps_to f univ`, like `has_image_in`, and refactor with that one. The very important thing though is not to have duplicates!